lib/jnpr/junos/utils/sw.py

```diff
@@ -430,11 +494,20 @@ def _progress(report):
                 _progress(
                     "validating software against current config,"
                     " please be patient ...")
-                v_ok = self.validate(remote_package, dev_timeout=timeout)
-                if v_ok is not True:
-                    return v_ok  # will be the string of output
+                v_ok = self.validate(remote_package, issu, dev_timeout=timeout)
 
-            if self._multi_RE is False:
+                if v_ok is not True:
+                    return v_ok
+
+            if issu is True:
```

Comment:
I am adding this. The plus point I saw by not putting that condition was even when the device was connected using gather_facts as False or facts gathering failed, this would work perfectly fine. I have added the check for now.